src/GraphControl/Control/Grapher.cpp

```diff
@@ -314,14 +314,18 @@ namespace GraphControl
                         initResult = nullopt;
                     }
                 }
+                else
+                {
+                    shouldKeepPreviousGraph = false;
```

Comment:
Can we remove the equation with an error and try to graph again?

Reply:
As discussed, this logic was already in place but was not triggering correctly, probably because the way equations get added was slightly changed at some point. Fixed now!